exercises/affine-cipher/.meta/src/reference/java/AffineCipher.java

```diff
@@ -0,0 +1,69 @@
+import java.util.ArrayList;
+import java.util.List;
+
+class AffineCipher {
+    private static final int GROUP_SIZE = 5;
+    private static final int ALPHABET_SIZE = 26;
+    private static final int ALPHABET_START_UNICODE = 97;
+
+    private static int modInverse(int keyA) {
+        int keyAMod = Math.floorMod(keyA, ALPHABET_SIZE);
+        for (int n = 1; n < ALPHABET_SIZE; n++) {
+            if (Math.floorMod(keyAMod * n, ALPHABET_SIZE) == 1) {
+                return n;
+            }
+        }
+        return 1;
+    }
+
+    private static String splitIntoFiveLetterWords(String text) {
+        List<String> words = new ArrayList<>();
+
+        for (int i = 0; i < text.length(); i += GROUP_SIZE) {
+            words.add(text.substring(i, Math.min(text.length(), i + GROUP_SIZE)));
+        }
+
+        return String.join(" ", words);
+    }
+
+    private static String translate(String text, int keyA, int keyB, int mode) {
```

Comment:
It looks like you're using `mode` essentially as a switch that should only have one of two values? In which case maybe using a boolean would be better, e.g. something like `shouldDecode`? Or you could use an enum? Not very important, but if there should only be two possible values then it's preferrable to use a type with only two values 🙂

Reply:
Great suggestion, thank you!

Reply:
🙂